authentication/src/main/java/io/camunda/authentication/CamundaOAuthPrincipalService.java

```diff
@@ -99,4 +109,14 @@ private String getUsernameFromClaims(final Map<String, Object> claims) {
                     "Configured username claim %s not found in claims. Please check your OIDC configuration."
                         .formatted(usernameClaim)));
   }
+
+  private String getApplicationIdFromClaims(final Map<String, Object> claims) {
+    return Optional.ofNullable(claims.get(applicationIdClaim))
+        .map(Object::toString)
```

Comment:
:question: Shouldn't we assert instead that it is actually a string? If it's accidentally a list, we'd just convert it to a string representation instead.